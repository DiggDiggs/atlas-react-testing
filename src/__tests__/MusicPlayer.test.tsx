// src/__tests__/MusicPlayer.test.tsx

import { render, screen, fireEvent } from "@testing-library/react";
import MusicPlayer from "../MusicPlayer.js"; // Adjust the import path as necessary

test("renders CurrentlyPlaying and Playlist components", () => {
  render(<MusicPlayer />);

  // Check if CurrentlyPlaying component is rendered
  expect(screen.getByText(/Currently Playing/i)).toBeInTheDocument();

  // Check if Playlist component is rendered
  expect(screen.getByText(/Playlist/i)).toBeInTheDocument();
});

test("initially displays the first song in the playlist", () => {
  render(<MusicPlayer />);

  // Check if the title of the first song is displayed
  expect(screen.getByText(/Painted in Blue/i)).toBeInTheDocument();

  // Check if the artist of the first song is displayed
  expect(screen.getByText(/Soul Canvas/i)).toBeInTheDocument();
});

test("plays the next song when playNext is called", () => {
  render(<MusicPlayer />);

  // Simulate playing the next song
  fireEvent.click(screen.getByText(/Next/i)); // Adjust this if there's no "Next" button

  // Check if the title of the second song is displayed
  expect(screen.getByText(/Tidal Drift/i)).toBeInTheDocument();
});

test("plays the previous song when playPrevious is called", () => {
  render(<MusicPlayer />);

  // Simulate playing the previous song
  fireEvent.click(screen.getByText(/Previous/i)); // Adjust this if there's no "Previous" button

  // Check if the title of the last song is displayed
  expect(screen.getByText(/Shatter the Silence/i)).toBeInTheDocument();
});

test("changes the current song when a playlist item is clicked", () => {
  render(<MusicPlayer />);

  // Simulate clicking on a playlist item
  fireEvent.click(screen.getByText(/Tidal Drift/i));

  // Check if the title of the clicked song is displayed
  expect(screen.getByText(/Tidal Drift/i)).toBeInTheDocument();
  expect(screen.getByText(/Echoes of the Sea/i)).toBeInTheDocument();
});
