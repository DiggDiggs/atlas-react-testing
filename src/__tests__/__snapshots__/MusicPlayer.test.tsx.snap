exports[`renders CurrentlyPlaying and Playlist components 1`] = `
<div>
  <div>
    <h2>Currently Playing</h2>
    <p>Song Title: Painted in Blue</p>
    <p>Artist: Soul Canvas</p>
    <button>Previous</button>
    <button>Next</button>
  </div>
  <div>
    <h2>Playlist</h2>
    <ul>
      <li>Painted in Blue - Soul Canvas</li>
      <li>Tidal Drift - Echoes of the Sea</li>
      <li>Fading Shadows - The Emberlight</li>
      <li>Cosmic Drift - Solar Flare</li>
      <li>Urban Serenade - Midnight Groove</li>
      <li>Whispers in the Wind - Rust & Ruin</li>
      <li>Electric Fever - Neon Jungle</li>
      <li>Edge of the Abyss - Steel Horizon</li>
      <li>Golden Haze - Velvet Waves</li>
      <li>Shatter the Silence - Thunderclap Echo</li>
    </ul>
  </div>
</div>
`;

exports[`initially displays the first song in the playlist 1`] = `
<div>
  <div>
    <h2>Currently Playing</h2>
    <p>Song Title: Painted in Blue</p>
    <p>Artist: Soul Canvas</p>
    <button>Previous</button>
    <button>Next</button>
  </div>
  <div>
    <h2>Playlist</h2>
    <ul>
      <li>Painted in Blue - Soul Canvas</li>
      <li>Tidal Drift - Echoes of the Sea</li>
      <li>Fading Shadows - The Emberlight</li>
      <li>Cosmic Drift - Solar Flare</li>
      <li>Urban Serenade - Midnight Groove</li>
      <li>Whispers in the Wind - Rust & Ruin</li>
      <li>Electric Fever - Neon Jungle</li>
      <li>Edge of the Abyss - Steel Horizon</li>
      <li>Golden Haze - Velvet Waves</li>
      <li>Shatter the Silence - Thunderclap Echo</li>
    </ul>
  </div>
</div>
`;

exports[`plays the next song when playNext is called 1`] = `
<div>
  <div>
    <h2>Currently Playing</h2>
    <p>Song Title: Tidal Drift</p>
    <p>Artist: Echoes of the Sea</p>
    <button>Previous</button>
    <button>Next</button>
  </div>
  <div>
    <h2>Playlist</h2>
    <ul>
      <li>Painted in Blue - Soul Canvas</li>
      <li>Tidal Drift - Echoes of the Sea</li>
      <li>Fading Shadows - The Emberlight</li>
      <li>Cosmic Drift - Solar Flare</li>
      <li>Urban Serenade - Midnight Groove</li>
      <li>Whispers in the Wind - Rust & Ruin</li>
      <li>Electric Fever - Neon Jungle</li>
      <li>Edge of the Abyss - Steel Horizon</li>
      <li>Golden Haze - Velvet Waves</li>
      <li>Shatter the Silence - Thunderclap Echo</li>
    </ul>
  </div>
</div>
`;

exports[`plays the previous song when playPrevious is called 1`] = `
<div>
  <div>
    <h2>Currently Playing</h2>
    <p>Song Title: Shatter the Silence</p>
    <p>Artist: Thunderclap Echo</p>
    <button>Previous</button>
    <button>Next</button>
  </div>
  <div>
    <h2>Playlist</h2>
    <ul>
      <li>Painted in Blue - Soul Canvas</li>
      <li>Tidal Drift - Echoes of the Sea</li>
      <li>Fading Shadows - The Emberlight</li>
      <li>Cosmic Drift - Solar Flare</li>
      <li>Urban Serenade - Midnight Groove</li>
      <li>Whispers in the Wind - Rust & Ruin</li>
      <li>Electric Fever - Neon Jungle</li>
      <li>Edge of the Abyss - Steel Horizon</li>
      <li>Golden Haze - Velvet Waves</li>
      <li>Shatter the Silence - Thunderclap Echo</li>
    </ul>
  </div>
</div>
`;

exports[`changes the current song when a playlist item is clicked 1`] = `
<div>
  <div>
    <h2>Currently Playing</h2>
    <p>Song Title: Tidal Drift</p>
    <p>Artist: Echoes of the Sea</p>
    <button>Previous</button>
    <button>Next</button>
  </div>
  <div>
    <h2>Playlist</h2>
    <ul>
      <li>Painted in Blue - Soul Canvas</li>
      <li>Tidal Drift - Echoes of the Sea</li>
      <li>Fading Shadows - The Emberlight</li>
      <li>Cosmic Drift - Solar Flare</li>
      <li>Urban Serenade - Midnight Groove</li>
      <li>Whispers in the Wind - Rust & Ruin</li>
      <li>Electric Fever - Neon Jungle</li>
      <li>Edge of the Abyss - Steel Horizon</li>
      <li>Golden Haze - Velvet Waves</li>
      <li>Shatter the Silence - Thunderclap Echo</li>
    </ul>
  </div>
</div>
`;